src/main/scala/io/citrine/lolo/bags/BaggedTrainingResult.scala

```diff
@@ -8,94 +8,78 @@ import scala.collection.parallel.immutable.ParSeq
 /** The result of training a [[Bagger]] to produce a [[BaggedModel]]. */
 sealed trait BaggedTrainingResult[+T] extends TrainingResult[T] {
 
-  override def getModel(): BaggedModel[T]
+  override def model: BaggedModel[T]
 }
 
 case class RegressionBaggerTrainingResult(
     ensembleModels: ParSeq[Model[Double]],
     Nib: Vector[Vector[Int]],
     trainingData: Seq[TrainingRow[Double]],
-    featureImportance: Option[Vector[Double]],
+    override val featureImportance: Option[Vector[Double]],
     biasModel: Option[Model[Double]] = None,
     rescaleRatio: Double = 1.0,
     disableBootstrap: Boolean = false
 ) extends BaggedTrainingResult[Double] {
 
   lazy val NibT: Vector[Vector[Int]] = Nib.transpose
-  lazy val model: BaggedRegressionModel =
-    BaggedRegressionModel(ensembleModels, Nib, rescaleRatio, disableBootstrap, biasModel)
-  lazy val predictedVsActual: Seq[(Vector[Any], Double, Double)] = trainingData.zip(NibT).flatMap {
-    case (TrainingRow(f, l, _), nb) =>
-      val oob = if (disableBootstrap) {
-        ensembleModels.zip(nb)
-      } else {
-        ensembleModels.zip(nb).filter(_._2 == 0)
-      }
-
-      if (oob.isEmpty || l.isNaN) {
-        Seq.empty
-      } else {
-        val predicted = oob.map(_._1.transform(Seq(f)).getExpected().head).sum / oob.size
-        Seq((f, predicted, l))
-      }
-  }
-
-  lazy val loss: Double = RegressionMetrics.RMSE(predictedVsActual)
-
-  override def getFeatureImportance(): Option[Vector[Double]] = featureImportance
 
-  override def getModel(): BaggedRegressionModel = model
-
-  override def getPredictedVsActual(): Option[Seq[(Vector[Any], Double, Double)]] = Some(predictedVsActual)
+  override lazy val model: BaggedRegressionModel =
+    BaggedRegressionModel(ensembleModels, Nib, rescaleRatio, disableBootstrap, biasModel)
 
-  override def getLoss(): Option[Double] = {
-    if (predictedVsActual.nonEmpty) {
-      Some(loss)
-    } else {
-      None
+  override lazy val predictedVsActual: Option[Seq[(Vector[Any], Double, Double)]] = Some(
+    trainingData.zip(NibT).flatMap {
+      case (TrainingRow(f, l, _), nb) =>
+        val oob = if (disableBootstrap) {
+          ensembleModels.zip(nb)
+        } else {
+          ensembleModels.zip(nb).filter(_._2 == 0)
+        }
+
+        if (oob.isEmpty || l.isNaN) {
+          Seq.empty
+        } else {
+          val predicted = oob.map(_._1.transform(Seq(f)).expected.head).sum / oob.size
+          Seq((f, predicted, l))
+        }
     }
+  )
+
+  override lazy val loss: Option[Double] = predictedVsActual.collect {
```

Comment:
This recreates the previous logic on `getLoss()`, which only returns a value if `predictedVsActual` is non-empty